plugins/nvme/fu-nvme-device.c

```diff
@@ -410,12 +410,14 @@ static gboolean
 fu_nvme_device_write_firmware (FuDevice *device, GBytes *fw, GError **error)
 {
 	FuNvmeDevice *self = FU_NVME_DEVICE (device);
+	g_autoptr(GBytes) fw2 = NULL;
 	g_autoptr(GPtrArray) chunks = NULL;
 	guint64 block_size = self->write_block_size > 0 ?
 			     self->write_block_size : 0x1000;
 
 	/* build packets */
-	chunks = fu_chunk_array_new_from_bytes (fw,
+	fw2 = fu_common_bytes_align (fw, block_size, 0xff);
```

Comment:
This basically throws garbage at the end of the payload to make it the right size for all chunks to be the same right?
Can you check with the other storage vendors on this?  I would like to make sure that it's not going to cause problems for some implementations to receive garbage in the last chunk (in which case this will need to be quirked for just this vendor)

Reply:
I looked at the other vendor on the LVFS you mentioned and the .bin file is perfectly aligned and so we don't add any extra garbage. I guess if you squint in a funny way the FWUG "_granularity_" could be taken to mean you have to manually pad the DWORDs to make each chunk the same size. It specifically mentions _aligning_ and _granularity_ and the latter I only know in terms of particle size. If you think a quirk is a safer option I'm okay with adding that too.

Reply:
Well just so far the other two vendors I've worked with have functioned fine "out of the box" and I don't want to see them break.  So I think either quirking it for this vendor or getting them to double check the implementation doesn't break them is how we should proceed.

Reply:
I've added this as a quirk.